virttest/libvirt_xml/domcapability_xml.py

```diff
@@ -32,6 +33,107 @@ def __init__(self, virsh_instance=base.virsh):
+                    for feature in mode_node.findall('feature'):
+                        item = {}
+                        item[feature.get('name')] = feature.get('policy')
+                        # Feature invtsc doesn't support migration, so not add it to feature list
+                        if 'invtsc' not in item:
+                            feature_list.append(item)
+        except AttributeError, elem_attr:
+            logging.warn("Failed to find attribute %s" % elem_attr)
+            feature_list = []
+        finally:
+            return feature_list
+
+    def get_hostmodel_name(self):
+        """
+        Get CPU modelname which explicitly specified by <model>.text
+        in cpu/mode[@name='host-model'] part of virsh domcapabilities.
+
+        In libvirt 3.9 and above, domcapabilities give the overall features supported
+        by current host/qemu/kvm via CPU model and a bunch of additional features. CPU
+        model is one set of features defined by libvirt in '/usr/share/libvirt/cpu_map.xml'
+        Specific features described in domcapabilities are addtional features, the valid
+        policy values are 'require' and 'disable', 'require' means the feature can be emulated
+        by host/qemu/kvm, 'disable' which means feature is disabled by current system due to
+        host CPU doesn't support or qemu/kvm has no ability to emulate it or some other
+        reasons like the feature block migration and so on.
+
+        Below is one snipped CPU host-model part output of domcapabilities.
+        <feature> tag specify addtional feature policy for one certain feature type.
+
+        virsh domcapabilities  | xmllint -xpath "/domainCapabilities/cpu/mode[@name='host-model']" -
+        <mode name="host-model" supported="yes">
+            <model fallback="forbid">Skylake-Client</model>
+            <vendor>Intel</vendor>
+            <feature policy="require" name="ss"/>
+            <feature policy="require" name="hypervisor"/>
+            <feature policy="require" name="tsc_adjust"/>
+            <feature policy="require" name="clflushopt"/>
+            <feature policy="require" name="pdpe1gb"/>
+            <feature policy="require" name="invtsc"/>
+        </mode>
+        Below is one snipped CPU host-model part output of domcapabilities.
+        feature tag spcific policy for one specific feature type.
+
+        virsh domcapabilities  | xmllint -xpath "/domainCapabilities/cpu/mode[@name='host-model']" -
+        <mode name="host-model" supported="yes">
+            <model fallback="forbid">Skylake-Client</model>
+            <vendor>Intel</vendor>
+            <feature policy="require" name="ss"/>
+        </mode>
+
+        :return: modelname string
+        """
+        xmltreefile = self.__dict_get__('xml')
+        try:
+            for mode_node in xmltreefile.findall('/cpu/mode'):
+                if mode_node.get('name') == 'host-model':
+                    return mode_node.find('model').text
```

Comment:
If xml have something like this
 ```
<cpu>
    <mode name='host-passthrough' supported='yes'/>
    <mode name='host-model' supported='no'/>
    <mode name='custom' supported='no'/>
  </cpu>
```
then we would get,

> WARNING:root:Failed to find attribute 'NoneType' object has no attribute 'text'

can we handle this too ?